lib/segment/src/vector_storage/query_scorer/custom_query_scorer.rs

```diff
@@ -98,6 +98,16 @@ impl<
         self.score(stored)
     }
 
+    fn score_stored_batch(&self, ids: &[PointOffsetType]) -> Vec<ScoreType> {
+        if check_ids_rather_contiguous(ids) {
```

Comment:
I seems to me that this check should happen on the storage level, cause it should know better how vectors are stored exactly.